packages/fx-core/templates/plugins/resource/botv2/bicep/function/functionConfiguration.template.bicep

```diff
@@ -0,0 +1,69 @@
+// Auto generated content, please customize files under provision folder
+
+@secure()
+param provisionParameters object
+param provisionOutputs object
+@secure()
+param currentAppSettings object
+
+var functionName = split(provisionOutputs.functionOutput.value.resourceId, '/')[8]
+
+{{#if (contains "fx-resource-aad-app-for-teams" plugins)}}
+var m365ClientId = provisionParameters['m365ClientId']
+  {{#if (contains "fx-resource-key-vault" plugins) }}
+var m365ClientSecret = \{{fx-resource-key-vault.References.m365ClientSecretReference}}
+  {{else}}
+var m365ClientSecret = provisionParameters['m365ClientSecret']
+  {{/if}}
+var m365TenantId = provisionParameters['m365TenantId']
+var m365OauthAuthorityHost = provisionParameters['m365OauthAuthorityHost']
+  {{#if (contains "fx-resource-frontend-hosting" plugins)}}
+    {{#if (contains "fx-resource-bot" plugins) }}
+var m365ApplicationIdUri = 'api://${ \{{fx-resource-frontend-hosting.References.domain}} }/botid-${botAadAppClientId}'
+    {{/if}}
+  {{else}}
+var m365ApplicationIdUri = 'api://botid-${botAadAppClientId}'
+  {{/if}}
+{{/if}}
+
+{{#if (contains "fx-resource-bot" plugins) }}
+var botAadAppClientId = provisionParameters['botAadAppClientId']
+  {{#if (contains "fx-resource-key-vault" plugins) }}
+var botAadAppClientSecret = \{{fx-resource-key-vault.References.botClientSecretReference}}
+  {{else}}
+var botAadAppClientSecret = provisionParameters['botAadAppClientSecret']
+  {{/if}}
+{{/if}}
+
+resource botFunctionSettings 'Microsoft.Web/sites/config@2021-02-01' = {
+  name: '${functionName}/appsettings'
+  properties: union({
+{{#if (contains "fx-resource-aad-app-for-teams" plugins)}}
+  {{#if (contains "dotnet" configurations)}}
+    TeamsFx__Authentication__ClientId: m365ClientId
+    TeamsFx__Authentication__ClientSecret: m365ClientSecret
+    TeamsFx__Authentication__OAuthAuthority: m365OauthAuthorityHost
+  {{/if}}
+  {{#if (contains "node" configurations)}}
+    INITIATE_LOGIN_ENDPOINT: uri(provisionOutputs.functionOutput.value.siteEndpoint, 'auth-start.html') // The page is used to let users consent required OAuth permissions during bot SSO process
```

Comment:
Maybe we could check with Bowen if we still need this config after they support non-SSO Tab. I am not sure if they provide user tutorial of Add SSO support on updating Bicep part